src/Acuminator/Acuminator.Analyzers/Analyzers/BQL/BqlParameterMismatch/ParametersCounterSymbolWalker.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Acuminator.Utilities;
+using PX.Data;
+
+
+namespace Acuminator.Analyzers
+{
+    public partial class BqlParameterMismatchAnalyzer : PXDiagnosticAnalyzer
+    {
+		/// <summary>
+		/// The BQL parameters counting symbol  walker.
+		/// </summary>
+		protected class ParametersCounterSymbolWalker : SymbolVisitor
+		{
+			private readonly SyntaxNodeAnalysisContext syntaxContext;
+            private readonly CancellationToken cancellationToken;
+
+			public ParametersCounter ParametersCounter { get; }
+
+			public ParametersCounterSymbolWalker(SyntaxNodeAnalysisContext aSyntaxContext, PXContext aPxContext)
```

Comment:
What does "a" prefix mean?

Reply:
argument. We have a recomenndation somewhere in style guide to use aParam for a class if you have param field and param parameter in constructor, Seems a little better to me than to use _param everywhere in class just because of constructor.

Reply:
First - not all style guides recommend this, it comes from visual basic where there is no case distinction for identifiers. Some guidelines do not recommend usage of prefix at all: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members.
If I remember correctly, our application code style too doesn't recommend it and suggets to use the aParameter name for constructor parameters. Even in .Net internals there is no consistent code style about this.
It is not a code style, it is a subject of holywar, actually:
https://www.c-sharpcorner.com/article/stop-use-var-everywhere-and-think-before-use-underscore-with-private-variable-in/
https://social.msdn.microsoft.com/Forums/vstudio/en-US/044a1246-0a15-48d5-902c-9c6c462af8b2/when-to-prefix-a-variable-with-underscore?forum=csharpgeneral

Imho, it seems clearer and doesn't clutter Intellisense.